### PR TITLE
PaneButton in Backpack

### PR DESCRIPTION
**Pull request by bencodeorg**
Wanted to get this out before I forget about it/it becomes outdated. Prefactor for https://github.com/code-dot-org/code-dot-org/pull/44341 -- moves the "Backpack" button that appears in the header of `JavalabEditor` to use `PaneHeader` like the other buttons in the header.

## Testing story

Tested manually.

**New (left) vs. old (right) -- only difference is in Backpack button:**

image

I am having a hard time telling if the backpack icon is appearing slightly lower than other icons -- I think it may be a pixel or two lower than it was previously, and the backpack button is slightly taller than it used to be (I think this is an improvement since it's now using the same component).

## Follow-up work

I think this should enabled us to merge the reverted centering of headers PR linked above.

### CODE REVIEW

File: apps/src/templates/PaneHeader.jsx [hunk 1]
```diff
@@ -186,7 +186,7 @@ export const PaneButton = Radium(function(props) {
       const Icon = icon.type;
       return (
         <Icon {...icon.props} style={{...iconStyle, ...icon.props.style}}>
-          {icon.children}
+          {icon.props.children}
```
Comment (bencodeorg) [inline apps/src/templates/PaneHeader.jsx:189]: I think this was just a typo -- I looked at the PR where this icon prop was added, and do not see anywhere that children would be used. I also checked all current uses of PaneButton -- most use iconClass, but few that use icon use a node that has no children.

**Comment by bencodeorg** [comment]
@code-dot-org/csa I forgot to put this out for review a couple of weeks ago. I'm [working on updating Backpack styling for this item](https://codedotorg.atlassian.net/browse/JAVA-430), so wanted to get eyes on this before we have merge conflicts. LMK if you think this refactor is a good idea!

**Review by molly-moen — APPROVED** [review]
LGTM!

**Review by jmkulwik — APPROVED** [review]
Nice change!

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice!

**Pull request merged**